### PR TITLE
【Hackathon No.55】 add channel_shuffle FP16/BF16 support and tests

### PR DESCRIPTION
**Pull request by superwinner1**
### PR types

Others
### PR changes

Others
### Describe

add channel_shuffle FP16/BF16 support and tests

### CODE REVIEW

**Comment by superwinner1** [comment]
@jzhang533 麻烦review~下，谢谢

**Comment by superwinner1** [comment]
@luotao1 麻烦review~一下，谢谢啦

**Comment by superwinner1** [comment]
@Vvsmile 麻烦review~下，谢谢啦

File: python/paddle/fluid/tests/unittests/test_channel_shuffle.py [hunk 1]
```diff
@@ -268,5 +272,53 @@ def error_data_format_layer():
...
+        if self.format == "NCHW":
+            shape = [n, c, h, w]
+        if self.format == "NHWC":
+            shape = [n, h, w, c]
+
+        groups = 3
+
+        x = np.random.random(shape).astype(self.dtype)
+        out = channel_shuffle_np(x, groups, self.format)
+        self.inputs = {'X': x}
```
Comment (Vvsmile) [inline python/paddle/fluid/tests/unittests/test_channel_shuffle.py]: BF16的单测中调用一下convert_float_to_uint16做一下类型转化

File: python/paddle/fluid/tests/unittests/test_channel_shuffle.py [hunk 2]
```diff
@@ -268,5 +272,53 @@ def error_data_format_layer():
...
+        self.attrs = {'groups': groups, "data_format": self.format}
+        self.outputs = {'Out': out}
```
Comment (Vvsmile) [inline python/paddle/fluid/tests/unittests/test_channel_shuffle.py]: BF16的单测中调用一下convert_float_to_uint16做一下类型转化

File: python/paddle/fluid/tests/unittests/test_channel_shuffle.py (hunk 1 above)
Comment (superwinner1) [reply python/paddle/fluid/tests/unittests/test_channel_shuffle.py]: done

File: python/paddle/fluid/tests/unittests/test_channel_shuffle.py (hunk 2 above)
Comment (superwinner1) [reply python/paddle/fluid/tests/unittests/test_channel_shuffle.py]: done

**Comment by superwinner1** [comment]
@Vvsmile ,麻烦再review~下，谢谢啦

**Comment by superwinner1** [comment]
@Vvsmile 辛苦review下，谢谢

**Review by ZzSean — APPROVED** [review]
LGTM

**Pull request merged**